apps/api/tests/test_jwt_auth.py

```diff
@@ -0,0 +1,221 @@
+from datetime import datetime, timedelta
+import json
+
+from django.conf import settings
+from django.test import RequestFactory
+
+import jwt
+from rest_framework.exceptions import AuthenticationFailed
+from rest_framework.response import Response
+from rest_framework_jwt.settings import api_settings
+
+from amo.tests import TestCase, WithDynamicEndpoints
+from apps.api.jwt_auth import handlers
+from apps.api.jwt_auth.views import JWTKeyAuthentication, JWTProtectedView
+from apps.api.models import APIKey, SYMMETRIC_JWT_TYPE
+from apps.users.models import UserProfile
+
+
+class ProtectedView(JWTProtectedView):
+    """
+    This is an example of a view that would be protected by JWT token auth.
+    """
+    def get(self, request):
+        return Response('some get response')
+
+    def post(self, request):
+        return Response({'user_pk': request.user.pk})
+
+
+class JWTAuthTester(TestCase):
+    fixtures = ['base/addon_3615']
+
+    def create_api_key(self, user, key='some-user-key',
+                       secret='some-shared-secret', **kw):
+        return APIKey.objects.create(type=SYMMETRIC_JWT_TYPE,
+                                     user=user, key=key, secret=secret,
+                                     **kw)
+
+    def auth_token_payload(self, user, issuer):
+        jwt_payload_handler = api_settings.JWT_PAYLOAD_HANDLER
+        return jwt_payload_handler(user, issuer)
+
+    def encode_token_payload(self, payload, secret):
+        jwt_encode_handler = api_settings.JWT_ENCODE_HANDLER
+        return jwt_encode_handler(payload, secret)
+
+    def create_auth_token(self, user, issuer, secret):
+        payload = self.auth_token_payload(user, issuer)
+        return self.encode_token_payload(payload, secret)
+
+
+class TestJWTProtectedView(WithDynamicEndpoints, JWTAuthTester):
+
+    def setUp(self):
+        super(TestJWTProtectedView, self).setUp()
+        self.endpoint(ProtectedView)
+
+    def request(self, method, *args, **kw):
+        handler = getattr(self.client, method)
+        return handler('/en-US/firefox/dynamic-endpoint', *args, **kw)
+
+    def login(self):
+        assert self.client.login(username='del@icio.us', password='password')
+        return UserProfile.objects.get(email='del@icio.us')
+
+    def jwt_request(self, token, method, *args, **kw):
+        kw['HTTP_AUTHORIZATION'] = 'JWT {}'.format(token)
```

Comment:
It looks like you could do this without mutating `kw` by passing this in as a named arg. It would blow up is `HTTP_AUTHORIZATION` is already in `kw` but that's probably what we want anyway. It's probably safe to mutate that `dict` but I still don't like mutating things.

```py
    def jwt_request(self, token, method, *args, **kw):
        return self.request(method, HTTP_AUTHORIZATION='JWT {}'.format(token),
                            *args, **kw)